src/main/java/org/opensearch/security/privileges/PrivilegesEvaluatorResponse.java

```diff
@@ -26,33 +26,61 @@
 
 package org.opensearch.security.privileges;
 
-import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashSet;
-import java.util.List;
 import java.util.Set;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 
 import org.opensearch.action.admin.indices.create.CreateIndexRequestBuilder;
 
 import com.selectivem.collections.CheckTable;
 
 public class PrivilegesEvaluatorResponse {
-    boolean allowed = false;
-    Set<String> missingSecurityRoles = new HashSet<>();
-    PrivilegesEvaluatorResponseState state = PrivilegesEvaluatorResponseState.PENDING;
-    CreateIndexRequestBuilder createIndexRequestBuilder;
-    private Set<String> onlyAllowedForIndices = ImmutableSet.of();
-    private CheckTable<String, String> indexToActionCheckTable;
+    private final boolean allowed;
+    private final CreateIndexRequestBuilder createIndexRequestBuilder;
+    private final ImmutableSet<String> onlyAllowedForIndices;
+    private final CheckTable<String, String> indexToActionCheckTable;
     private String privilegeMatrix;
-    private String reason;
+    private final String reason;
 
     /**
      * Contains issues that were encountered during privilege evaluation. Can be used for logging.
      */
-    private List<PrivilegesEvaluationException> evaluationExceptions = new ArrayList<>();
+    private ImmutableList<PrivilegesEvaluationException> evaluationExceptions;
+
+    public PrivilegesEvaluatorResponse(
```

Comment:
out of curiosity, why not implement a complete Builder pattern over here? we already use chaining for method calls for this class.